assets/src/js/Components/ArchivePicker/SingleSidebar.js

```diff
@@ -111,15 +113,34 @@ export default function SingleSidebar({image}) {
           )}
           {view === EDITOR_VIEW && (
             <>
-              {!image.wordpress_id ? (
+              {isUserSelectingFeaturedImage && image.wordpress_id && (
+                <button
+                  disabled={!!processing}
+                  className="button sidebar-action"
+                  onClick={async () => await setCustomFeaturedImage(image.wordpress_id)}
+                >
+                  {__('Set Featured Image', 'planet4-master-theme-backend')}
+                </button>
+              )}
+              {isUserSelectingFeaturedImage && !image.wordpress_id && (
+                <button
+                  disabled={!!processing}
+                  className="button sidebar-action"
+                  onClick={async () => await includeInWp([image.id], view, true)}
+                >
+                  {__('Import to Library & Set Featured Image', 'planet4-master-theme-backend')}
+                </button>
+              )}
+              {!isUserSelectingFeaturedImage && !image.wordpress_id && (
                 <button
                   disabled={!!processing}
                   className="button sidebar-action"
                   onClick={async () => await includeInWp([image.id], view)}
                 >
                   {__('Import to Library & Post', 'planet4-master-theme-backend')}
                 </button>
-              ) : (
+              )}
+              {!isUserSelectingFeaturedImage && image.wordpress_id && (
                 <button
```

Comment:
This if condition can be also added into the above condition wrapped by `Fragments`.

Reply:
Yea, like I said earlier, I created a follow up [ticket](https://jira.greenpeace.org/browse/PLANET-7641) to refactor this components.